kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/JSONInputFormat.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data.spi.filesystem;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.kitesdk.compat.Hadoop;
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.spi.DataModelUtil;
+import org.kitesdk.data.spi.EntityAccessor;
+
+class JSONInputFormat<E> extends FileInputFormat<E, Void> {
+  private DatasetDescriptor descriptor = null;
+  private EntityAccessor<E> accessor = null;
+
+  public void setDescriptor(DatasetDescriptor descriptor) {
+    this.descriptor = descriptor;
+  }
+
+  public void setType(Class<E> type) {
+    this.accessor = DataModelUtil.accessor(type, descriptor.getSchema());
+  }
+
+  @Override
+  protected boolean isSplitable(JobContext context, Path filename) {
+    return false;
```

Comment:
It would be base don the format of the file. I can see at least three common file formats, there might be more:

1. Newline delimited JSON objects without newlines internal to the objects
2. Whitespace delimited JSON objects with embedded newlines possible
3. A list of JSON objects where the top-level JSON element is an array

Format (1) is splittable on newlines. What' I'm proposing is we could add that as an option now as it would hit a good number of use cases. Format (2) is a more general version of format (1) and the `} {` trick might work for both, but as you said we haven't implemented it yet.

Format (3) may never be splittable as it will use commas in between records (`[ {...}, {...}, ...]`). If we want/need to support that format as well then we'll have to be able to configure the file format we're reading.